dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/TaskInstanceMapper.java

```diff
@@ -81,7 +81,7 @@ IPage<TaskInstance> queryTaskInstanceListPaging(IPage<TaskInstance> page,
      * @param endTime endTime
      * @return task instance
      */
-    TaskInstance queryLastTaskInstance(@Param("taskCode") long taskCode, @Param("startTime") Date startTime, @Param("endTime") Date endTime);
+    TaskInstance queryLastTaskInstance(@Param("taskCode") long taskCode, @Param("processInstanceId") long processInstanceId);
```

Comment:
When relying on specific tasks, I think that the execution of tasks in the time cycle has nothing to do with which workflow instance, so this issue cannot be modified in this way

Reply:
because taskInstance don't keep the `scheduleTime`, but processInstance do, if only use start_time, and processInstance start with a scheduleTime, the dependent node will use `scheduleTime` to calculate a `bizdate` to find the task, it may can't detect the task in original `bizdate`.

the actual situation I have encountered is:
etl task `B` processes a zipper table, it must run continuously by date, it's previews dependent task `D`  depends on `A[T]` and `B[T-1](self yesterday)`, in some reason, `A` blocked until the `next day`, so `B.startTime` is `[T+1]`, and next day `D` will use `bizdate[T]` calculated by `scheduleTime[T+1]`  to finding `B[T]`, there is no `B` started on time `[T]`, that maks 
 dependent task `D[T+1]` will never stop, every time this happens, I need to manually stop the processInstance, and set the state of `D` to `Forced Success`.

In fact taskInstance can only run attached to a processInstance, so I think it is right to find taskInstance through processInstance, and in 2.0.5, it did in the same way.

Reply:
I see what you mean. I think there were bugs before 2.0.5. The execution of tasks in the time cycle has nothing to do with which workflow instance. A task should be finished yesterday, but it finish in today, is there a problem ? And it should be solved manually

Reply:
Yes, I understand.
usually data processing is required to be completed on the same day,
but there are other tasks that need to be performed continuously every day and not sensitive to date delays.
upstream data may have acceptable latency, when the delayed data is released,
at that time, we usually expect the scheduling platform to automatically and continuously process the delayed data (including tasks that are blocked downstream for many days), especially when some third-party data access is involved.

Some scheduling platforms have the feature of automatic date flop of tasks, especially this feature can also be implemented through the `dependent task` of Dolphinscheduler.

most important is, this update changed some rules, i can never use `complement` command to rerun the delayed process which include `dependent task`(it never worked with complement), that why i submited this pr.

Reply:
> A task should be finished yesterday

This is just one of the cases, what if the task is allowed to finish in the next days?